ui/ui_statusbar.c

```diff
@@ -13,111 +13,122 @@
 #include "display/scope.h"
 
 
-uint32_t ui_statusbar_info(char *buf)
+uint32_t ui_statusbar_info(char *buf, size_t buffLen)
 {
     uint32_t len=0;
     uint32_t temp=0;
     uint32_t cnt=0;
 
-	len+=ui_term_color_text_background_buf(&buf[len], 0x000000, BP_COLOR_GREY);
+	len+=ui_term_color_text_background_buf(&buf[len], buffLen, 0x000000, BP_COLOR_GREY);
+    buffLen = sizeof(tx_sb_buf) - len;	
     
     if(system_config.psu)
     {
-        temp=sprintf(&buf[len],"Vout: %u.%uV",
+        temp=snprintf(&buf[len], buffLen, "Vout: %u.%uV",
            (system_config.psu_voltage)/10000, ((system_config.psu_voltage)%10000)/100
         );
         len+=temp;
         cnt+=temp;
+        buffLen = sizeof(tx_sb_buf) - len;	
         if(system_config.psu_current_limit_en)
         {
-            temp=sprintf(&buf[len],"/%u.%umA max",
+            temp=snprintf(&buf[len], buffLen, "/%u.%umA max",
                 (system_config.psu_current_limit)/10000, ((system_config.psu_current_limit)%10000)/100 
             );
             len+=temp;
-            cnt+=temp;           
+            cnt+=temp;
+            buffLen = sizeof(tx_sb_buf) - len;	

```

Comment:
remove this line.  Don't override the `buffLen` variable ... the caller provides the buffer pointer, so the caller provides the length of that buffer.   Overriding the value is a fragile piece of code, and in this case, it's also wrong in most places in this file..  Fragility example: The caller could legitimately change later to use a different buffer,  and this would break.

Reply:
Review changes, adjust accordingly, re-submit PR is recommended.

Reply:
Please re-review changes.  Thanks again!